lmnet/lmnet/datasets/tensorflow_datasets.py

```diff
@@ -0,0 +1,141 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 The Blueoil Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+import numpy as np
+import tensorflow as tf
+import tensorflow_datasets as tfds
+
+from lmnet.datasets.base import Base, ObjectDetectionBase
+from lmnet import data_processor
+
+
+class TensorFlowDatasetsBase(Base):
+    """
+    Abstract dataset class for loading TensorFlow Datasets.
+    The parameter "extend_dir" should be a path to the prepared data of tfds.
+    "extend_dir" can be a path either in local or in Google Cloud Storage.
+    """
+    available_subsets = ["train", "train_validation_saving", "validation"]
```

Comment:
Why do you fix `available_subsets` as `["train", "train_validation_saving", "validation"]` ? This doesn't include `test` subset.

Reply:
Thanks. I'll fix it.

Reply:
This variable is defined as `@staticmethod` in the base class. But available subsets will be decided after the class is instantiated in case of this class because it depends on what dataset will be loaded.

So I'll define this variable as `["train", "validation"]` in this class because these two subsets are always available.

Reply:
> So I'll define this variable as ["train", "validation"] in this class because these two subsets are always available.

Nice 👍